### ISSUE TITLE
linter: `eslint/no-unused-vars` with `args: 'none'` reports unused rest bindings `...args`

### ISSUE THREAD

**Issue by jeanne-cai** [oxc-project/oxc#22625]
### What version of Oxlint are you using?

1.66.0

### What command did you run?

`yarn oxlint -c oxlint.config.mts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
    categories: {
        correctness: 'off',
    },
    rules: {
        'eslint/no-unused-vars': [
            'error',
            {
                args: 'none',
            },
        ],
    },
});
```

### What happened?

With `eslint/no-unused-vars` configured as `args: 'none'`, oxlint still reports **unused rest parameters** (`...args`). However, ESLint using `@typescript-eslint/no-unused-vars` with the same configuration does not report them.

To reproduce:
```ts
// Regular unused param: correctly ignores
export function foo(unused: string, used: string) {
    return used;
}

// Unused rest param `...args`: oxlint still reports it, while ESLint does not.
export function rest(...args: unknown[]) {
    return 1;
}
```

Output:

```bash
  × eslint(no-unused-vars): Parameter 'args' is declared but never used.
   ╭─[test.ts:7:25]
 6 │ // Unused rest param `...args`: oxlint still reports it, while ESLint does not.
 7 │ export function rest(...args: unknown[]) {
   ·                         ──┬─
   ·                           ╰── 'args' is declared here
 8 │     return 1;
   ╰────
  help: Consider removing this parameter.

Found 0 warnings and 1 error.
Finished in 102ms on 1 file with 1 rules using 16 threads.
```
